src/rastervision/ml_backends/tf_object_detection_api.py

```diff
@@ -327,16 +337,28 @@ class TFObjectDetectionAPI(MLBackend):
     def __init__(self):
         self.detection_graph = None
 
-    def convert_training_data(self, training_data, validation_data, class_map,
-                              options):
+    def per_project_data_processor(self, project, data, class_map, options):
+
         training_package = TrainingPackage(options.output_uri)
+        tf_examples = make_tf_examples(data, class_map)
+        record_path = training_package.get_local_path(
+            training_package.get_record_uri(project.id))
+        write_tf_record(tf_examples, record_path)
```

Comment:
When this function exits, `training_package` is deleted which deletes the temporary directory it created, which deletes `record_path`. This is a problem because when `all_projects_dataset_processor` is executed, the paths in `training_data` no longer exist.

Reply:
Ah! I've never tried output_uri being remote... I'll have to experiment...